### PR TITLE
Chore: first pass at new *Transform.operation() naming

### PR DESCRIPTION
**Pull request by niedzielski**
- Module or class has type in the name. e.g., RedLinkTransform,
  ReferenceTransform. I've also taken the liberty of dropping the
  plural on RedLinks since a more terse style was wanted and the other
  transforms were singular.

- Rename some of the class / module methods to be more terse. This still
  needs a bit of work. e,g., RedLinkTransform.hide(),
  ReferenceTransform.opacity(.5). I changed
  CollapseTableTransform.collapseTables() to just .transform() since
  what it really does is enable tables to be collapsed and the default
  state happens to be collapsed.

- ElementUtilties looked like it was aready in pretty good shape.

- Add . prefix to method test cases.

Since this is a large and very breaking patch. Please focus on
eliminating unwanted changes and fixing consistency bugs rather than
requesting features and enhancements.

todo: update changelog with list of API changes once agreed on.

### CODE REVIEW

**Comment by niedzielski** [comment]
@montehurd, please review as soon as you can. This has a ton of the breaking API changes wanted.

File: src/transform/CollapseTableTransform.js [hunk 1]
```diff
@@ -177,24 +177,24 @@ const newCaption = (title, headerText) => {
  * @param {?string} infoboxTitle
  * @param {?string} otherTitle
  * @param {?string} footerTitle
- * @param {?FooterDivClickCallback} footerDivClickCallback
+ * @param {?FooterCallback} FooterCallback
  * @return {void}
  */
-const collapseTables = (window, content, pageTitle, isMainPage, infoboxTitle, otherTitle,
-  footerTitle, footerDivClickCallback) => {
+const transform = (window, content, pageTitle, isMainPage, infoboxTitle, otherTitle,
```
Comment (montehurd) [inline src/transform/CollapseTableTransform.js]: Can this be `collapse`?

File: src/transform/CollapseTableTransform.js [hunk 2]
```diff
@@ -108,7 +108,7 @@ const toggleCollapseClickCallback = function(footerDivClickCallback) {
  * @param {!HTMLElement} table
  * @return {!boolean} true if table should be collapsed, false otherwise.
  */
-const shouldTableBeCollapsed = table => {
+const collapsible = table => {
```
Comment (montehurd) [inline src/transform/CollapseTableTransform.js]: Can this be `isCollapsible`?

File: src/transform/index.js [hunk 3]
```diff
@@ -1,12 +1,12 @@
...
-import WidenImage from './WidenImage'
+import RedLinkTransform from './RedLinkTransform'
+import WidenImageTransform from './WidenImageTransform'
 
 export default {
-  CollapseTable,
-  RedLinks,
-  WidenImage,
+  CollapseTableTransform,
+  RedLinkTransform,
```
Comment (montehurd) [inline src/transform/index.js]: `HideRedLinkTransform` would match the naming scheme used by the transforms above and below.

**Comment by montehurd** [comment]
@niedzielski Hey sorry I didn't have a chance to look at this earlier! I left a couple quick comments.

File: src/transform/CollapseTableTransform.js (hunk 1 above)
Comment (niedzielski) [inline src/transform/CollapseTableTransform.js]: This does more than just collapse the tables though. It's a transformation that enables tables that can collapse or expand and collapse just happens to be the default state.

File: src/transform/index.js (hunk 3 above)
Comment (niedzielski) [inline src/transform/index.js]: Thanks, I missed this. Are you cool with the redundancy of "hide" in `HideRedLinkTransform.hide()` then?

**Comment by niedzielski** [comment]
Either as part of this PR or a new PR, please rename all camelCase and shish-kebab CSS symbols to snake_case.

**Comment by niedzielski** [comment]
I'm willing to reconsider isolating each transform + helper classes to a distinct subdirectory in light of the size of the lazy load and footer transforms. We'll need to decide on a "every directory has an index.js" vs some redundancy in the module names.

**Comment by montehurd** [comment]
Closing for now - added note to circle back and re-review later.

**Pull request closed without merging**